### PR TITLE
Add zsh completion

### PR DESCRIPTION
**Pull request by haraldkubota**
This is (so far) just the zsh completion file. No integration with autoconf, automake or making packages. Mainly because I've never used any of those tools and I likely will mess them up.

### CODE REVIEW

**Comment by haraldkubota** [comment]
Anyone who could review this?

**Review by smcv — CHANGES_REQUESTED** [review]
To be useful, this will need to be installed, something like this (untested):


```
# in configure.ac
AC_ARG_WITH([zsh-completion-dir],
    AS_HELP_STRING([--with-zsh-completion-dir[=PATH]],
        [Install the zsh auto-completion script in this directory. @<:@default=yes@:>@]),
    [],
    [with_zsh_completion_dir=yes])

AS_IF([test "x$with_zsh_completion_dir" = "xyes"],
    [ZSH_COMPLETION_DIR='$datadir/zsh/site-functions"])
AC_SUBST([ZSH_COMPLETION_DIR])                                                   
AM_CONDITIONAL([ENABLE_ZSH_COMPLETION], [test "x$with_zsh_completion_dir" != "xno"])
```

```
# in Makefile.am
if ENABLE_ZSH_COMPLETION
zshcompletiondir = $(ZSH_COMPLETION_DIR
dist_zshcompletion_DATA = completions/zsh/_bwrap
endif
```

Just after the similar code for bash completions would probably be a good place to put these.

Try it with:

```
./autogen.sh
make
make install DESTDIR="$(pwd)/DESTDIR"
```

If all goes well you should find that the zsh completion is installed to `DESTDIR/usr/local/share/zsh/site-functions/_bwrap`.

**Comment by smcv** [comment]
I don't really know zsh completion syntax, and we should check through the options and make sure they match what's implemented (particularly if any of my MRs that add new options get merged), but this looks really useful - thanks!

File: completions/zsh/_bwrap [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#compdef bwrap
+
+_all_caps() {
+    # $ egrep '#define\sCAP_\w+\s+[0-9]+' /usr/include/linux/capability.h | awk '{print $2}' | xargs echo
```
[the remark this answers is not in the document]
Comment (haraldkubota) [reply completions/zsh/_bwrap]: Yes, all_caps is defined once and then immediate used once.
I could not find a simple way to gather all kernel capabilities dynamically.

egrep...yeah, old school SysV user here.

[the remark this answers is not in the document]
Comment (haraldkubota) [reply completions/zsh/_bwrap]: Do I have to click on "Resolve conversation"?

File: completions/zsh/_bwrap [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    echo CAP_CHOWN CAP_DAC_OVERRIDE CAP_DAC_READ_SEARCH CAP_FOWNER CAP_FSETID \
+    CAP_KILL CAP_SETGID CAP_SETUID CAP_SETPCAP CAP_LINUX_IMMUTABLE \
+    CAP_NET_BIND_SERVICE CAP_NET_BROADCAST CAP_NET_ADMIN CAP_NET_RAW \
+    CAP_IPC_LOCK CAP_IPC_OWNER CAP_SYS_MODULE CAP_SYS_RAWIO CAP_SYS_CHROOT \
+    CAP_SYS_PTRACE CAP_SYS_PACCT CAP_SYS_ADMIN CAP_SYS_BOOT CAP_SYS_NICE \
+    CAP_SYS_RESOURCE CAP_SYS_TIME CAP_SYS_TTY_CONFIG CAP_MKNOD CAP_LEASE \
+    CAP_AUDIT_WRITE CAP_AUDIT_CONTROL CAP_SETFCAP CAP_MAC_OVERRIDE \
+    CAP_MAC_ADMIN CAP_SYSLOG CAP_WAKE_ALARM CAP_BLOCK_SUSPEND CAP_AUDIT_READ
```
[the remark this answers is not in the document]
Comment (haraldkubota) [reply completions/zsh/_bwrap]: Removed already

**Review by haraldkubota — COMMENTED** [review]
Oh, I didn't realize that I can do the review!
Reviewed and cross-checked with the output of brwap --help and the options are not only in the same order as the command completions, but I cannot find anything extra or anything missing.

**Review by smcv — COMMENTED** [review]
Looks generally good, although I haven't tested it yet.

**Review by smcv — APPROVED** [review]
Seems to work nicely!

**Comment by smcv** [comment]
@cgwalters, are you considering me to be enough of a maintainer that I can merge other people's branches (e.g. this one), or do you want power of veto?

**Comment by cgwalters** [comment]
> @cgwalters, are you considering me to be enough of a maintainer that I can merge other people's branches (e.g. this one)

You're *at least* equal and have been doing more work, so please don't block on me!

(I skimmed this and it looks sane FWIW)

**Pull request merged**